app/ui/src/Components.elm

```diff
@@ -0,0 +1,33 @@
+module Components exposing (..)
+
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick, onInput, onSubmit)
+import Model exposing (Output(..))
+
+
+layout : List (Html msg) -> Html msg
+layout children =
+    div [] children
+
+
+query : (String -> msg) -> msg -> Html msg
+query queryChange submit =
+    Html.form [ onSubmit submit ]
+        [ label [ for "query" ] [ text "Your query: " ]
+        , input [ id "query", onInput queryChange ] []
+        , button [ type_ "submit", onClick submit ] [ text "Update" ]
+        ]
+
+
+output : Output -> Html msg
```

Comment:
Right, in that case it should be `ApiResponse (List String)` and `Status (String)`, Output then being `ApiResponse | Status`. 

In your code snippet above `[message]` is ambiguous. That can be _either_ a status message _or_ a server response with a single message in it. 

Maybe I should just rename the variants to `Messages` and `Status` then?